### ISSUE TITLE
[Scala] Serialization exception when a class contains a List

### ISSUE THREAD

**Issue by ghostdogpr** [alipay/fury#1137]
**Describe the bug**
If you try to serialize a class that contains a List, it fails with an error like this:
```
Exception in thread "main" java.lang.ClassCastException: class io.fury.serializer.ReplaceResolveSerializer cannot be cast to class io.fury.serializer.collection.AbstractCollectionSerializer (io.fury.serializer.ReplaceResolveSerializer and io.fury.serializer.collection.AbstractCollectionSerializer are in unnamed module of loader 'app')
	at Reproducer_CFuryRefCodec_457312648_716083600_1064154107.writeCollectionClassInfo$(Reproducer_CFuryRefCodec_457312648_716083600_1064154107.java:67)
	at Reproducer_CFuryRefCodec_457312648_716083600_1064154107.write(Reproducer_CFuryRefCodec_457312648_716083600_1064154107.java:134)
	at io.fury.Fury.writeData(Fury.java:518)
	at io.fury.Fury.writeRef(Fury.java:311)
	at io.fury.Fury.write(Fury.java:286)
	at io.fury.Fury.serialize(Fury.java:231)
	at io.fury.Fury.serialize(Fury.java:185)
	at Reproducer$.delayedEndpoint$Reproducer$1(Testt.scala:15)
	at Reproducer$delayedInit$body.apply(Testt.scala:3)
	at scala.Function0.apply$mcV$sp(Function0.scala:39)
	at scala.Function0.apply$mcV$sp$(Function0.scala:39)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
	at scala.App.$anonfun$main$1(App.scala:76)
	at scala.App.$anonfun$main$1$adapted(App.scala:76)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:919)
	at scala.App.main(App.scala:76)
	at scala.App.main$(App.scala:74)
	at Reproducer$.main(Testt.scala:3)
	at Reproducer.main(Testt.scala)
```

**Expected behavior**
There should be no such error.

**To Reproduce**
```scala
import io.fury.Fury

object Reproducer extends App {

  val fury = Fury
    .builder()
    .withScalaOptimizationEnabled(true)
    .requireClassRegistration(false)
    .withRefTracking(true)
    .build()

  case class C(list: List[Int])

  val p = C(List(1))
  println(fury.deserialize(fury.serialize(p)))
}
```

**Environment (please complete the following information):**
Tried with Fury 0.3.1 and 0.4.0-SNAPSHOT, Scala 2.13.

**Comment by chaokunyang** [comment]
Hi @ghostdogpr , thanks for finding this bug and report to us, I fixed it in #1140. Could you try it out again? If Fury works for your scenerios without further issues. We can release a new scala version ASAP.

**Comment by chaokunyang** [comment]
@ghostdogpr Thanks for point out this, I'll try it later and create a new fix
